commit 9004a9ef832311af98e63471beed23aece744acd (refs/changes/02/702/1)
Author: Pedro P. Candel <pedro@joyent.com>
Date:   2016-10-17T14:34:10+02:00 (3 years ago)
    
    TOOLS-1575 sdcadm platform install does not exit non-zero on failure

diff --git a/lib/platform.js b/lib/platform.js
index 49e7804..a4298e7 100644
--- a/lib/platform.js
+++ b/lib/platform.js
@@ -734,18 +734,23 @@ Platform.prototype.install = function install(opts, callback) {
                 progress('After correcting above problem, rerun ' +
                          '`sdcadm platform install %s`.', filepath);
             }
-            return updateHistory(hist, callback);
+            updateHistory(hist, function (_histErr) {
+                callback(err);
+            });
+            return;
         }
 
         if (deleteOnFinish) {
-            return cleanup(callback);
+            cleanup(callback);
+            return;
         } else {
             progress('Platform image explicitly specified; ' +
                      'will not delete %s', filepath);
             progress('Installation complete');
         }
 
-        return updateHistory(hist, callback);
+        updateHistory(hist, callback);
+        return;
     });
 };
 
